docs/scripts/prepare_notebooks_for_ci.py

```diff
@@ -0,0 +1,181 @@
+"""Preprocess notebooks for CI. Currently adds VCR cassettes and optionally removes pip install cells."""
+
+import json
+import logging
+import os
+
+import click
+import nbformat
+
+logger = logging.getLogger(__name__)
+NOTEBOOK_DIRS = ("docs/docs/tutorials",)
+DOCS_PATH = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+CASSETTES_PATH = os.path.join(DOCS_PATH, "cassettes")
+
+# TODO: populate if needed
+NOTEBOOKS_NO_CASSETTES = [
+    "docs/docs/tutorials/retrievers.ipynb",  # TODO: fix non-determinism
+]
+
+NOTEBOOKS_NO_EXECUTION = [
+    "docs/docs/tutorials/graph.ipynb",  # Requires local graph db running
+    "docs/docs/tutorials/local_rag.ipynb",  # Local LLMs
```

Comment:
ooc why can't this be run off a cassette?

Reply:
Local LLM notebook I think may not make http calls, there's just significant overhead running local models (e.g., ollama). LMK if I've misunderstood.